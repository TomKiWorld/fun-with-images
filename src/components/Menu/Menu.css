.menu-container {
  position: relative;
}

.menu-container .menu { 
  content: 'â‰¡';
  color: #CC0B78;
  font-size: 6rem;
  margin: 0 2rem;
}

.menu-container .menu-links ul {
  text-align: left;
  list-style: none;
  margin: 0;
  padding: 1rem;
}

.menu-container .nav-drop-down {
  position: absolute;
  top: 3.5rem;
  right: 4rem;
  width: 11rem;
  font-size: 1rem;
  background-color: rgba(255, 255, 255, 0.5);
  box-shadow: 4px 4px 8px 0 rgba( 0, 0, 0, .2 );
}

.menu-container .nav-drop-down li {
  padding: 0.5rem;
}

.menu-container .nav-drop-down ul li:hover,
.menu-containern .nav-drop-down ul li:focus {
  background-color: #FFF;
}

.menu-container .nav-open {
  display: none;
}

@media screen and (min-width: 37rem) {
  .menu-container .menu {
    display: none;
  }

  .menu-container .nav-open {
    display: block;
  }

  .menu-container .nav-open ul {
    display: flex;
  }
}
