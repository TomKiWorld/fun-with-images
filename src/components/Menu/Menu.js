import React, { Component } from 'react';

import './Menu.css';

class Menu extends Component {
  constructor(props) {
    super(props);
    this.state = {
      dropdownOpen: false
    };
  }

  toggle = (status) => {
    if (status) {
      this.setState({dropdownOpen: status})
    } else {
      this.setState(prevState => ({
        dropdownOpen: !prevState.dropdownOpen
      }));
    }
  }

  showDropDown = () => {
    const { onRouteChange  } = this.props;
    if (this.state.dropdownOpen) {
      return (
        <div 
          className='nav-drop-down menu-links'
          onMouseLeave={() => this.toggle(false)}
           >
            <ul>
              <li 
                onClick={() => onRouteChange('home')} >Home</li>
              <li 
                onClick={() => onRouteChange('profile')} >View Profile</li>
              <li 
                onClick={() => onRouteChange('signin')} >Sign out</li>
            </ul>
        </div>
      )
    }
  }

  render() {
    const { onRouteChange  } = this.props;
    return (      
      <div className='menu-container tc'>
        <div aria-expanded={this.state.dropdownOpen}>
          <p
            className='menu'
            onClick={this.toggle}
            >â‰¡</p>
        </div>
        {this.showDropDown()}
        <div 
          className='nav-open menu-links'
           >
            <ul>
              <li 
                onClick={() => onRouteChange('home')}
                className='link dim black underline pa3 pointer' >Home</li>
              <li 
                onClick={() => onRouteChange('profile')}
                className='link dim black underline pa3 pointer' >View Profile</li>
              <li 
                onClick={() => onRouteChange('signin')}
                className='link dim black underline pa3 pointer' >Sign out</li>
            </ul>
        </div>
      </div>
    )
  }
}

export default Menu;
