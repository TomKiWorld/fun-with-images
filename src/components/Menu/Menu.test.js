import { shallow } from 'enzyme';
import React from 'react';
import Menu from './Menu';

describe('Menu Component', () => {
  const wrapper = shallow(<Menu />);

  it('Expect to render Menu Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
