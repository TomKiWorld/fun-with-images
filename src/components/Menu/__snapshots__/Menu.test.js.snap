// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component Expect to render Menu Component 1`] = `ShallowWrapper {}`;
